fiftyone/core/fields.py

```diff
@@ -514,10 +514,7 @@ def to_python(self, value):
         if value is None:
             return None
 
-        # Explicitly converting to UTC is important here because PyMongo loads
-        # everything as `datetime`, which will respect `fo.config.timezone`,
-        # but we always need UTC here for the conversion back to `date`
-        return value.astimezone(pytz.utc).date()
+        return value.date()
```

Comment:
I did not understand this comment, why would we want it to be in UTC to get the date?
Removing this conversion fixes the issue described in PR.